tests/test_pipelines/test_demultiplexing.py

```diff
@@ -24,8 +24,24 @@ def test_run(self):
         fake_fastq_pairs = [
             [('L1_R1_001.fastq.gz', 'L1_R2_001.fastq.gz')], [('L2_R1_001.fastq.gz', 'L2_R2_001.fastq.gz')]
         ]
+        fake_genome_response = {
+            "_updated": "30_11_2018_15:13:43",
+            "assembly_name": "phix174",
+            "analyses_supported": ["qc"],
+            "data_source": "",
+            "_links": {"self": {"title": "genome", "href": "genomes/phix174"}},
+            "_etag": "175b41e3909a93a8298ac1d5d4dfc7292df4b580",
+            "data_files": {"fasta": "/path/to/phix.fa"},
+            "_created": "30_11_2018_15:13:43",
+            "species": "PhiX",
+            "genome_size": 5386,
+            "_id": "5c0153a716a5772f9e9cfdcc"
+        }
         patch_run_crawler = patch('analysis_driver.pipelines.demultiplexing.RunCrawler', autospec=True)
         patch_find = patch('egcg_core.util.find_all_fastq_pairs', side_effect=fake_fastq_pairs)
+        patch_get_document = patch('egcg_core.rest_communication.get_document',
+                                   return_value=fake_genome_response)
+        patch_get_document.start()
```

Comment:
This is a slightly different technique than the one you two currently use, which I found online. Let me know if you have any objections.

Reply:
using `start()` and `stop()` only makes sense if the patch is started and stopped in different functions.
otherwise you should use `with`
There are no advantages to use `start()` and `stop()` here

Reply:
I was not sure how to patch it with "with" since this function is not explicitly used in test code. I will experiment with using "with" again to be sure of the difficulty I was facing.